code/processes/wdb.q

```diff
@@ -311,44 +324,46 @@ merge:{[dir;pt;tableinfo;mergelimits;hdbsettings;mergemethod]
   ] 
  };	
 	
-endofdaymerge:{[dir;pt;tablist;mergelimits;hdbsettings;mergemethod]		
+endofdaymerge:{[dir;pt;tablist;mergelimits;hdbsettings;mergemethod]
   /- merge data from partitons
-  $[(0 < count .z.pd[]) and ((system "s")<0);
-    [.lg.o[`merge;"merging on worker"];
-     {(neg x)(`.wdb.reloadsymfile;y);(neg x)(::)}[;.Q.dd[hdbsettings `hdbdir;`sym]] each .z.pd[];
-     /-upsert .merge.partsize data to sort workers, only needed for part and hybrid method 
-     if[(mergemode~`hybrid)or(mergemode~`part);
-       {(neg x)(upsert;`.merge.partsizes;y);(neg x)(::)}[;.merge.partsizes] each .z.pd[];
-       ];
-     merge[dir;pt;;mergelimits;hdbsettings;mergemethod] peach flip (key tablist;value tablist);
-     /-clear out in memory table, .merge.partsizes, and call sort worker processes to do the same
-     .lg.o[`eod;"Delete from partsizes"];
-     delete from `.merge.partsizes;
-     {(neg x)({.lg.o[`eod;"Delete from partsizes"];
-               delete from `.merge.partsizes;
-               /- run a garbage collection if enabled
-               if[gc;.gc.run[]]};`);(neg x)(::)} each .z.pd[];
-    ];	
-    [.lg.o[`merge;"merging on main"];
-     reloadsymfile[.Q.dd[hdbsettings `hdbdir;`sym]];
-     merge[dir;pt;;mergelimits;hdbsettings;mergemethod] each flip (key tablist;value tablist);
-     .lg.o[`eod;"Delete from partsizes"];
-     delete from `.merge.partsizes;
-    ]
-   ];
-  /- if path exists, delete it
-  if[not () ~ key savedir; 
-    .lg.o[`merge;"deleting temp storage directory"];
-    .os.deldir .os.pth[string[` sv savedir,`$string[pt]]];
-    ];
-  /-call the posteod function
-  .save.postreplay[hdbsettings[`hdbdir];pt];
-  $[permitreload; 
-    doreload[pt];
-    if[gc;.gc.run[]];
-    ];
-  };
-	
+  /- .z.pd funciton in finspace will cause an error. Add in this flag to skip over the use of .z.pd. This should be temporary and will be removed when issue resolved by AWS.
+  if[not .finspace.enabled;
```

Comment:
does this skip the entire EOD merge when finspace is enabled? is that what's desired?

If so, probably better to just return early if it's enabled e.g.

```q
  if[.finspace.enabled;:()];
```

and leave the rest of function unchanged

Reply:
It does not skip the entire eodmerge function. It skips over the first if statement where .z.pd is used. So the function essentially starts at the comment "If path exists, delete it"

Reply:
right, but it's inside that first statement that the actual `merge` function gets called, so we're skipping the merge, right?

Reply:
Oh sorry yes i see, it does skip over the merge.